src/vmm/src/dumbo/tcp/handler.rs

```diff
@@ -57,25 +57,25 @@ pub enum WriteEvent {
 ///
 /// [`receive_packet`]: struct.TcpIPv4Handler.html#method.receive_packet
 /// [`TcpIPv4Handler`]: struct.TcpIPv4Handler.html
-#[derive(Debug, PartialEq, Eq, derive_more::From)]
+#[derive(Debug, PartialEq, Eq, thiserror::Error, displaydoc::Display)]
 pub enum RecvError {
     /// The inner segment has an invalid destination port.
     InvalidPort,
-    /// The handler encountered an error while parsing the inner TCP segment.
-    TcpSegment(TcpSegmentError),
+    /// The handler encountered an error while parsing the inner TCP segment: {0}
+    TcpSegment(#[from] TcpSegmentError),
 }
 
 /// Describes errors which may be encountered by the [`write_next_packet`] method from
 /// [`TcpIPv4Handler`].
 ///
 /// [`write_next_packet`]: struct.TcpIPv4Handler.html#method.write_next_packet
 /// [`TcpIPv4Handler`]: struct.TcpIPv4Handler.html
-#[derive(Debug, PartialEq, Eq, derive_more::From)]
+#[derive(Debug, PartialEq, Eq, thiserror::Error, displaydoc::Display)]
 pub enum WriteNextError {
     /// There was an error while writing the contents of the IPv4 packet.
-    IPv4Packet(IPv4PacketError),
+    IPv4Packet(#[from] IPv4PacketError),
```

Comment:
do we not need `{0}` for these?

Reply:
I think it is worth having `Display` to be implemented to all errors.

Reply:
yeah, my point is would we like to add that into the message (currently it is missing)?

Reply:
Oh, yeah I totally missed it. Thx. This is exactly the issue I also want to avoid. Each error must include internal data to the message.

Reply:
fixed now